tools/bwa.wdl

```diff
@@ -223,11 +226,11 @@ task bwa_mem {
         File read_one_fastq_gz
         File bwa_db_tar_gz
         File? read_two_fastq_gz
-        String prefix = sub(
+        String prefix = sub(sub(
             basename(read_one_fastq_gz),
             "([_\\.][rR][12])?(\\.subsampled)?\\.(fastq|fq)(\\.gz)?$",
             ""
-        )
+        ), "\\.([rR][12])\\.", ".")
```

Comment:
Can you give me an example `read_one_fastq_qz` to this? I want to make sure I'm understanding this correctly. I hope we can refactor this to be clearer. Or if we can't simplify the code this could get a comment.

Reply:
We could probably simplify this. `"([_\\.][rR][12])?(\\.subsampled)?\\.(fastq|fq)(\\.gz)?$"` is the original regex that we use everywhere. The issue here is that there is a split number(e.g. `00000` between `R1` and `.fastq.gz`). 


Here is an example pair of FASTQ files. The first regex removes the `.fastq.gz` extension. The second specifically looks for the `.R1.` and `.R2.` strings. 
```
SJALL019091_G1.Exome.r585684.R1.000000.fastq.gz
SJALL019091_G1.Exome.r585684.R2.000000.fastq.gz
```

Reply:
So we start with `SJALL019091_G1.Exome.r585684.R1.000000.fastq.gz` and want to end with `SJALL019091_G1.Exome.r585684.000000` which will have `.bam` appended to it?

Reply:
Yes

Reply:
```
        String prefix = sub(
            sub(
                basename(read_one_fastq_gz),
                "(fastq|fq)(\\.gz)?$",
                ""
            ),
            "\\.([rR][12])\\.",
            "."
        )
```

Reply:
Do we want to trim this logic down here, in the `bwa` WDL? There could still be `subsampled` input and for a workflow that doesn't use `split_fastq` there wouldn't likely be a split counter.\

This would still work and support both cases, right?
```
        String prefix = sub(
            sub(
                basename(read_one_fastq_gz),
                "(\\.subsampled)?\\.(fastq|fq)(\\.gz)?$",
                ""
            ),
            "\\.([rR][12])\\.",
            "."
        )
```

Reply:
True, maybe this isn't the place for this logic change? I can see one of two changes here:
1. Replace all instances of this regex with the "double `sub()`" workshopped here
    - I think your suggestion is good, but we've dropped support for `_R1` suffixes. So `"\\.([rR][12])\\.",` should become something like `[_\\.][rR][12]`
2. Leave this bit alone, reverting all changes made, and instead sort out the naming from the workflow call

Reply:
I've gone with #2. I think that's the best approach.